redbox-core/redbox/graph/nodes/processes.py

```diff
@@ -280,17 +293,17 @@ def _tool(state: RedboxState) -> dict[str, Any]:
 
                 # Deal with InjectedState
                 args = tool_call["args"].copy()
+                log.info(f"Invoking tool {tool_call['name']} with args {args}")
                 if has_injected_state(tool):
                     args["state"] = state
 
-                log.info(f"Invoking tool {tool_call['name']} with args {args}")
-
                 # Invoke the tool
                 try:
                     result_state_update = tool.invoke(args) or {}
                     tool_called_state_update = {"tool_calls": {tool_id: {"called": True, "tool": tool_call}}}
                     state_updates.append(result_state_update | tool_called_state_update)
                 except Exception as e:
+                    raise e
```

Comment:
why catch this only to immediately raise it?

Reply:
ah good catch, we were debugging. Removed now